pkg/server/api/localauthority/v1/service.go

```diff
@@ -0,0 +1,279 @@
+	GetNextJWTKeySlot() manager.Slot
+	PrepareJWTKey(ctx context.Context) error
+	RotateJWTKey()
+
+	// X509
+	GetCurrentX509CASlot() manager.Slot
+	GetNextX509CASlot() manager.Slot
+	PrepareX509CA(ctx context.Context) error
+	RotateX509CA()
+}
+
+// Config is the service configuration
+type Config struct {
+	TrustDomain spiffeid.TrustDomain
+	DataStore   datastore.DataStore
+	CAManager   CAManager
+}
+
+// New creates a new LocalAuthority service
+func New(config Config) *Service {
+	return &Service{
+		td: config.TrustDomain,
+		ds: config.DataStore,
+		ca: config.CAManager,
+	}
+}
+
+// Service implements the v1 LocalAuthority service
+type Service struct {
+	localauthorityv1.UnsafeLocalAuthorityServer
+
+	td spiffeid.TrustDomain
+	ds datastore.DataStore
+	ca CAManager
+}
+
+func (s *Service) GetJWTAuthorityState(context.Context, *localauthorityv1.GetJWTAuthorityStateRequest) (*localauthorityv1.GetJWTAuthorityStateResponse, error) {
+	return nil, status.Error(codes.Unimplemented, "unimplemented")
+}
+
+func (s *Service) PrepareJWTAuthority(context.Context, *localauthorityv1.PrepareJWTAuthorityRequest) (*localauthorityv1.PrepareJWTAuthorityResponse, error) {
+	return nil, status.Error(codes.Unimplemented, "unimplemented")
+}
+
+func (s *Service) ActivateJWTAuthority(context.Context, *localauthorityv1.ActivateJWTAuthorityRequest) (*localauthorityv1.ActivateJWTAuthorityResponse, error) {
+	return nil, status.Error(codes.Unimplemented, "unimplemented")
+}
+
+func (s *Service) TaintJWTAuthority(context.Context, *localauthorityv1.TaintJWTAuthorityRequest) (*localauthorityv1.TaintJWTAuthorityResponse, error) {
+	return nil, status.Error(codes.Unimplemented, "unimplemented")
+}
+
+func (s *Service) RevokeJWTAuthority(context.Context, *localauthorityv1.RevokeJWTAuthorityRequest) (*localauthorityv1.RevokeJWTAuthorityResponse, error) {
+	return nil, status.Error(codes.Unimplemented, "unimplemented")
+}
+
+func (s *Service) GetX509AuthorityState(ctx context.Context, _ *localauthorityv1.GetX509AuthorityStateRequest) (*localauthorityv1.GetX509AuthorityStateResponse, error) {
+	log := rpccontext.Logger(ctx)
+
+	var states []*localauthorityv1.AuthorityState
+	current := s.ca.GetCurrentX509CASlot()
+	if !current.IsEmpty() {
+		if current.AuthorityID() == "" {
+			return nil, api.MakeErr(log, codes.Internal, "current slot does not contains authority ID", nil)
+		}
+
+		states = append(states, &localauthorityv1.AuthorityState{
+			AuthorityId: current.AuthorityID(),
+			Status:      localauthorityv1.AuthorityState_ACTIVE,
+		})
+	}
+
+	next := s.ca.GetNextX509CASlot()
+	// when next has a key indicates that it was initialized
+	if next.AuthorityID() != "" {
+		status := localauthorityv1.AuthorityState_PREPARED
+		// CA is removed from slot on rotation
+		if next.IsEmpty() {
+			status = localauthorityv1.AuthorityState_OLD
+		}
+
+		states = append(states, &localauthorityv1.AuthorityState{
+			AuthorityId: next.AuthorityID(),
+			Status:      status,
+		})
+	}
```

Comment:
Actually we have an State field on journal now, and we can just parse that status into a  `localauthorityv1.AuthorityState` I was thinking on do that on this PR but I choose to keep it out for now, 
About the third `slot` approach, are you ok with creating another issue for that? 
I can start working on that and we can adapt this code if required, I `think` since it is an ID we'll not affect key managers but I will need to review that,

Reply:
> are you ok with creating another issue for that?

Yes for sure! Didn't intend for all that to be in scope for this PR, just that it seems like a good future direction